opentelemetry-rails/lib/opentelemetry/rails/railtie.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+module OpenTelemetry
+  module Rails
+    # Railtie automatically configure the OpentTelemetry SDK
+    #
+    # This railtie will set defaults for the following environment variables:
+    # `OTEL_SERVICE_NAME` - if unset, will default to the Rails application name
+    # `OTEL_RESOURCE_ATTRIBUTES` - if unset, will set the deployment environment to the current Rails environment
+    #
+    # This will also set the `OpenTelemetry.logger` to the Rails logger.
+    class Railtie < ::Rails::Railtie
+      railtie_name :opentelemetry
+      config.opentelemetry = ActiveSupport::OrderedOptions.new
+      initializer 'opentelemetry.configure' do |app|
```

Comment:
kinda scope creep but how do we feel about wrapping this in an `on/off` environment variable?

If the goal is "just update the gemfile once and control everything with env vars" it might be nice to have a break-glass env-var option to completely disable instrumentation.